exercises/scale-generator/canonical-data.json

```diff
@@ -0,0 +1,181 @@
+{
+  "exercise": "scale-generator",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "comments": [
+        "These tests have no interval.",
+        "Chromatic scale is considered the default scale"
+      ],
+      "description": "Chromatic scales",
+      "cases": [
+        {
+          "description": "Chromatic scale with sharps",
+          "property": "pitches",
+          "input": {
+            "tonic": "C"
+          },
+          "expected": [ "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" ]
+        },
+        {
+          "description": "Chromatic scale with flats",
+          "property": "pitches",
+          "input": {
+            "tonic": "F"
+          },
+          "expected": [ "F", "Gb", "G", "Ab", "A", "Bb", "B", "C", "Db", "D", "Eb", "E" ]
+        }
+      ]
+    },
+    {
+      "comments": [
+        "These tests all have intervals and are explorations of different",
+        "traversals of the scale."
+      ],
+      "description": "Scales with specified intervals",
+      "cases": [
+        {
+          "description": "Simple major scale",
+          "comments": [
+            "The simplest major scale, with no sharps or flats."
+          ],
+          "property": "pitches",
+          "input": {
+            "tonic": "C",
+            "intervals": "MMmMMMm"
+          },
+          "expected": [ "C", "D", "E", "F", "G", "A", "B" ]
+        },
+        {
+          "description": "Major scale with sharps",
+          "property": "pitches",
+          "input": {
+            "tonic": "G",
+            "intervals": "MMmMMMm"
+          },
+          "expected": [ "G", "A", "B", "C", "D", "E", "F#" ]
+        },
+        {
+          "description": "Major scale with flats",
+          "property": "pitches",
+          "input": {
+            "tonic": "g",
```

Comment:
Here we see `g` being used to depict a major scale with flats. Most code that precisely follows the README and respects the `tonic` and `intervals` specified **will** in fact give the expected answer.

Note that there would be no way to specify this "g major" in verbal communication, since any utterance that sounds like "gee major" will be interpreted as "G major". In addition, the table at the bottom of https://en.wikipedia.org/wiki/Key_(music) specifies that G major will always have sharps, not flats; and that the `g` is only used to represent g minor. I thus contend that this test case is correct in code but is inconsistent with music theory.

If wishing to use a major scale that has flats *and* is consistent with music theory, it seems it would be better to use one under the "Flats" heading of the table. Perhaps F major would be good.

I will express no preference on whether what I have stated means it is better to keep this test case or change it. So in GitHub terms I will neither Approve nor Request Changes on the current state; of course someone else may feel free to do so.

We can also contend that the fact that it is possible to specify a pair of inputs that makes no sense `tonic g, intevals MMmMMMm` is a bug in the specification, and that would be fair to say, but this PR is not obligated to correct the specification.

Reply:
F major makes a lot more sense. Will fix.

Reply:
Fixed.